eZ/Publish/API/Repository/TransactionService.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\API\Repository;
+
+use Closure;
+
+interface TransactionService
+{
+    /**
+     * Begin transaction.
+     *
+     * Begins an transaction, make sure you'll call commit or rollback when done,
+     * otherwise work will be lost.
+     */
+    public function beginTransaction(): void;
+
+    /**
+     * Commit transaction.
+     *
+     * Commit transaction, or throw exceptions if no transactions has been started.
+     *
+     * @throws \RuntimeException If no transaction has been started
+     */
+    public function commit(): void;
+
+    /**
+     * Rollback transaction.
+     *
+     * Rollback transaction, or throw exceptions if no transactions has been started.
+     *
+     * @throws \RuntimeException If no transaction has been started
+     */
+    public function rollback(): void;
+
+    /**
+     * Executes a function in a transaction.
+     *
+     * If an exception occurs during execution of the function or transaction commit,
```

Comment:
Did you mean the following?
"If an exception is thrown while a function is executed or a transaction is committed,"

Reply:
It's "the" function. Specifically it's the function provided as callback that is wrapped with transaction.